### ISSUE TITLE
Implement dynamic coordinates lookup and store the result in the database

### ISSUE THREAD

**Issue by jevhen-ponomarenko** [JankatKaan/CityDistanceCalculatorAPI#1]
Let's add some complexity to your solution....

In you current solution the database is [basically this variable](https://github.com/JankatKaan/CityDistanceCalculatorAPI/blob/84b16137f368a4f584ca8d0d683c89814119d34d/WebApplication2/Controllers/DistanceController.cs#L10). So we are going to move this data to an actual database. 

1. Install some database engine on you system (locally or in Docker for extra points). You can follow a [guide](https://learn.microsoft.com/en-us/windows/wsl/tutorials/wsl-database) on how to do that. 
2. Connect you app to the database. 
3. Design your schema and use it to save/retrieve `City` records to/from the database.

In order to allow users to write any city they desire, even the one that's not in the database, we need to use 3rd party API to get the coordinates. Implement this using any API you want to, there is plenty that are free to some extent. 
- https://docs.maptiler.com/cloud/api/geocoding/ . You can create a free account here and use the geocoding api. 

the algorithm will now be:

1. User searches for distance betweeen two cities
2. find the cities in the database
  3. if the city is not found in the dataabase, look it up using API
  4. if it's found, return the coordinates
5. Calculate and return the distance between cities.

Please try to finish it in the minimal fashion: don't spend too much time cleaning the input etc. I just want to see that you are able to implement this as a whole. The details don't matter. 

If you have any questions, please leave a comment to this issue. 
  
  Useful links:
  - https://learn.microsoft.com/en-us/windows/wsl/install
  - https://www.c-sharpcorner.com/article/building-a-powerful-asp-net-core-web-api-with-postgresql/

**Comment by JankatKaan** [comment]
I will consider your suggestions. Thank you for providing an explanation and links. I'll send you updates shortly.

**Comment by jevhen-ponomarenko** [comment]
I would really appreciate if you push your solution and create a Pull Request at least 2 hours prior to the scheduled meeting tomorrow.

**Comment by JankatKaan** [comment]
I developed an API based on your suggestions and used Docker on the WSL installation. For the database, I used Postgres. Initially, my cities table was empty. If a requested city cannot be found locally, the API obtains the coordinates from a third-party API (Maptiler) and then saves the coordinates to the database for later use. This method is efficient in saving new cities and reducing requests to third-party APIs. Further requests are faster using this method.
